mysql/mysql.go

```diff
@@ -44,12 +45,37 @@ type URLOpener struct {
 
 // OpenMySQLURL opens a new database connection wrapped with OpenCensus instrumentation.
 func (uo *URLOpener) OpenMySQLURL(_ context.Context, u *url.URL) (*sql.DB, error) {
+	cfg, err := ConfigFromURL(u)
+	if err != nil {
+		return nil, err
+	}
 	return sql.OpenDB(connector{
-		dsn:       strings.TrimPrefix(u.String(), Scheme+"://"),
+		dsn:       cfg.FormatDSN(),
 		traceOpts: append([]ocsql.TraceOption(nil), uo.TraceOpts...),
 	}), nil
 }
 
+// ConfigFromURL creates a mysql.Config from URL.
```

Comment:
IIUC, this is a straight copy from the previous location, right? And same with the test below?

Reply:
Yes